src/components/treemap.tsx

```diff
@@ -80,19 +80,48 @@ export default class Treemap extends React.Component<
       width={d.x1 - d.x0}
       height={d.y1 - d.y0}
     />
-  ));
+  ))
+
+    sizingProperties = d => {
+        const [ minValue, maxValue ] = this.props.fontSize;
+        const x = max([0, Math.floor(((d.x1 - d.x0) - 8))]);
+        const y = max([0, Math.floor(((d.y1 - d.y0) - 8))]);
+        const label = d.data.location_name;
+        const width = stringWidth(label, 12) / 12;
+        const direction = (y > x && (width * (minValue + 1)) > x) ? 'topBottom' : 'leftRight';
+        return { x, y, width, direction, minValue, maxValue };
+    }
+
+    fontSize = d => {
+        const { x, y, width, direction, minValue, maxValue } = this.sizingProperties(d);
```

Comment:
Unless you have an aversion to it, I think listing these vertically would be easier to add to. It also follows the team's current convention more. I know the airbnb lint rules yell at me when I list more than one.